scripts/install.sh

```diff
@@ -0,0 +1,224 @@
+#!/bin/bash
+
+ROOT_DIR="$(pwd)"
+
+if [[ "$OSTYPE" == "darwin"* ]]; then
+    OS="macOS"
+    NUMBER_OF_PROCESSORS="$(sysctl -n hw.ncpu)"
+fi
+
+if [[ "$OSTYPE" == "linux-gnu"* ]]; then
+    OS="Linux"
+    NUMBER_OF_PROCESSORS="$(nproc)"
+fi
+
+GENERATOR="Unix Makefiles"
+
+SUBMODULE_DIR="${ROOT_DIR}/vendor"
+SUBMODULE_INCLUDE_DIR="${SUBMODULE_DIR}/include"
+SUBMODULE_LIB_DIR="${ROOT_DIR}/lib/${OS}"
+
+echo "Installing vulkan dependencies..."
+
+update_submodules() {
+    git submodule update --init ${SUBMODULE_DIR}/$1
+}
+
+mkdirp() { 
+    mkdir -p $1 
+}
+
+setup_volk() {
+    echo "Cloning Volk..."
+    update_submodules volk
+
+    mkdirp ${SUBMODULE_DIR}/include/volk
+
+    cp ${SUBMODULE_DIR}/volk/volk.h ${SUBMODULE_INCLUDE_DIR}/volk 
+    cp ${SUBMODULE_DIR}/volk/volk.c ${SUBMODULE_INCLUDE_DIR}/volk
+}
+
+setup_glfw() {
+    echo "Cloning glfw..."
+
+    update_submodules glfw
+
+    cd ${SUBMODULE_DIR}/glfw ; cmake -G "${GENERATOR}" . ; make -j${NUMBER_OF_PROCESSORS}
+
+    mkdirp ${SUBMODULE_LIB_DIR}
+
+    cp ${SUBMODULE_DIR}/glfw/src/libglfw3.a ${SUBMODULE_LIB_DIR}
+
+    cd ${ROOT_DIR}
```

Comment:
if we can avoid uses of `cd`, a failed script won't dump us somewhere we weren't expecting